UI design for class page

Arshiya-rze opened the pull request.

Arshiya-rze committed:
Update class models and services and update app.routes for adding edit/ create class

Arshiya-rze committed:
Implement design for class list and card components

Arshiya-rze committed:
Add create class form with jalali datepicker

Arshiya-rze committed:
Implement edit class page with toggleable edit mode

Arshiya-rze committed:
Cleanup and removed persian comments

FaranakBaran commented on client/src/app/models/class.model.ts:
```diff
@@ -0,0 +1,34 @@
+  professorNames: string[];
+}
+
+export interface AddClass {
```
chera inja az ClassBase estefade nakardi ke faghat "courseName" va "siteName" ro inja behesh ezafe koni. Intori:

export interface AddClass extends Omit<ClassBase, 'days'> {
  courseName: string; 
  siteName: string;
}

Arshiya-rze committed:
refactor(models): optimize AddClass interface inheritance

Refactored AddClass to extend ClassBase using Omit instead of duplicating properties.
This reduces code redundancy and addresses the PR review comment regarding model structure.

FaranakBaran commented:
<img width="585" height="1266" alt="IMG_7703" src="https://github.com/user-attachments/assets/bfc259ad-5e81-4836-987a-af3f1f752dd4" />

Arshiya-rze committed:
fix(class-create): initialize missing properties in class creation payload

The updated AddClass interface made 'isStarted' and 'isActive' required properties.
Added these fields to the createClass payload to resolve the TypeScript compilation error.

FaranakBaran reviewed (approved):
Done

Arshiya-rze committed:
fix(class-ui): correct site name property usage in templates

Updated class-create and class-card templates to access 'site.name' instead of 'site.siteName'.
This resolves the build error caused by a property name mismatch with the Site model.
